File: python/tdd/book.py
import enum

class Type(enum.Enum):
    HARDBACK = 1
    PAPERBACK = 2

class Book:
    def __init__(self, title, author):
        self.title = title
        self.author = author
        self.recomended_retail_price = None
        self.type = None
        self.months_since_publication = None

    def get_recomended_retail_price(self):
        return self.recomended_retail_price

    def set_recomended_retail_price(self, recomended_retail_price):
        self.recomended_retail_price = recomended_retail_price

    def get_type(self):
        return self.type

    def set_type(self, type_):
        self.type = type_

    def get_months_since_publication(self):
        return self.months_since_publication

    def set_months_since_publication(self, months_since_publication):
        self.months_since_publication = months_since_publication

File: csharp/TDD/Book.cs
public enum BookType { Hardback, Paperback };

public class Book
{
    public string Title { get; }
    public string Author { get; }
    public double RecommendedRetailPrice { get; set; }
    public BookType Type { get; set; }
    public int MonthsSincePublication { get; set; }

    public Book(string title, string author)
    {
        Title = title;
        Author = author;
    }
}
